### PR TITLE
Add a base test case for IJ plugin components that supports service mocking

### PR DESCRIPTION
**Pull request by patflynn**
…ocking

### CODE REVIEW

**Comment by patflynn** [comment]
@elharo @loosebazooka 

I had to create a new PR for some reason. Your comments were incorporated.

File: common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      fail("Could not serialize object: " + e.getMessage());
+    } finally {
+      if (out != null) {
+        try {
+          out.close();
+        } catch (IOException e) {
+          // ignore
+        }
+      }
+      try {
```
Comment (elharo) [inline common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java]: unnecessary; out.close closes the chained stream

Comment (patflynn) [inline common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java]: fixed

**Comment by patflynn** [comment]
any more comments?

File: common-test-lib/build.gradle [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apply plugin: 'java'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compile files('third_party/junit/junit-4.11.jar')
```
Comment (loosebazooka) [inline common-test-lib/build.gradle]: compile or testCompile?

Comment (loosebazooka) [inline common-test-lib/build.gradle]: just kidding

**Comment by loosebazooka** [comment]
lgtm?

**Comment by elharo** [comment]
lgtm

**Pull request merged**